test/framework/autoscaler_helpers.go

```diff
@@ -0,0 +1,426 @@
+	// Each pod should requests memory resource of about 60% of the node capacity so that at most one pod
+	// fits on each node. Setting a replicas of workers + 1 would ensure we have pods that cannot be scheduled.
+	// This will force exactly one extra node to be spun up.
+	replicas := workers + 1
+	memoryRequired := int64(float64(memory.Value()) * 0.6)
+	podMemory := resource.NewQuantity(memoryRequired, resource.BinarySI)
+
+	scalelUpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "scale-up",
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				"app": "scale-up",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(int32(replicas)),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "scale-up",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "scale-up",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "busybox",
+							Image: "busybox",
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: *podMemory,
+								},
+							},
+							Command: []string{"/bin/sh", "-c", "echo \"up\" & sleep infinity"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	By("Scale up deployment created")
+	Expect(input.ClusterProxy.GetClient().Create(ctx, scalelUpDeployment)).To(Succeed(), "failed to create the scale up pod")
+
+	By("Wait for the scale up deployment to become ready (this implies machines to be created)")
+	WaitForDeploymentsAvailable(ctx, WaitForDeploymentsAvailableInput{
+		Getter:     input.ClusterProxy.GetClient(),
+		Deployment: scalelUpDeployment,
+	}, intervals...)
+}
+
+type ProcessYAMLInput struct {
+	Template             []byte
+	ClusterctlConfigPath string
+	Env                  map[string]string
+}
+
+func ProcessYAML(input *ProcessYAMLInput) ([]byte, error) {
+	for n, v := range input.Env {
+		_ = os.Setenv(n, v)
+	}
+
+	c, err := clusterctlclient.New(input.ClusterctlConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	options := clusterctlclient.ProcessYAMLOptions{
+		ReaderSource: &clusterctlclient.ReaderSourceOptions{
+			Reader: bytes.NewReader(input.Template),
+		},
+	}
+
+	printer, err := c.ProcessYAML(options)
+	if err != nil {
+		return nil, err
+	}
+
+	out, err := printer.Yaml()
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+type DisableAutoscalerForMachineDeploymentTopologyAndWaitInput struct {
+	ClusterProxy                  ClusterProxy
+	Cluster                       *clusterv1.Cluster
+	WaitForAnnotationsToBeDropped []interface{}
+}
+
+// DisableAutoscalerForMachineDeploymentTopologyAndWait drop the autoscaler annotations from the MachineDeploymentTopology
+// and waits till the annotations are dropped from the underlying MachineDeployment.
```

Comment:
Let's also add a comment that it verifies that after we dropped the annotations the replica field values are preserved